### ISSUE TITLE
Incorrect MeshDataCache behavior with FreeCamera script

### ISSUE THREAD

**Issue by pacyfist** [FlaxEngine/FlaxEngine#932]
I'm currently using `dotnet7` branch of the flax engine, but same issue was observed on different branches including `master`

I have prepared a script that uses MeshDataCache to count vertices and triangles after loading the model.

```csharp
using FlaxEngine;
using FlaxEngine.Utilities;

namespace Game
{
    public class LoadModel : Script
    {
        public Model CubeModel;

        public override void OnStart()
        {
            CubeModel.WaitForLoaded();

            var cache = new MeshDataCache();
            cache.RequestMeshData(CubeModel);
            cache.Finished += () =>
            {
                var vertices_base = cache.MeshDatas[0][0].VertexBuffer;
                var triangles_base = cache.MeshDatas[0][0].IndexBuffer;

                FlaxEngine.Debug.Log($"LOAD Base Vertices Count = {vertices_base.Length}");
                FlaxEngine.Debug.Log($"LOAD Base Triangles Count = {triangles_base.Length}");
            };
        }
    }

}
```

My scene contains only two actors.

1. A standard issue Camera
2. An freshly added EmptyActor

I have attached the script to the EmptyActor, and configured a custom cube model to be passed into it.

![obraz](https://user-images.githubusercontent.com/2545509/216850466-3ed769a5-98cf-4223-8562-77b307588faa.png)

Now running the game in editor and as cooked executable result in the logs are same, and perfectly correct.

```text
[ 00:00:03.077 ]: [Info] LOAD Base Vertices Count = 24
[ 00:00:03.077 ]: [Info] LOAD Base Triangles Count = 36
```

But as soon as I add the standard issue FreeCamera script (or my custom camera script) to the camera strange things happen.

In editor all is fine and everything works as expected.

```text
[ 00:20:15.355 ]: [Info] Created graph for scene 'Scene' in 0 ms
[ 00:20:15.355 ]: [Info] Scene loaded in 1 ms
[ 00:20:15.356 ]: [Info] LOAD Base Vertices Count = 24
[ 00:20:15.356 ]: [Info] LOAD Base Triangles Count = 36
[ 00:20:15.357 ]: [Info] [PlayMode] Enter
[ 00:20:17.586 ]: [Info] [PlayMode] Stop
```

but as cooked game there be dragons.

```text
[ 00:00:02.998 ]: [Info] Loading the first scene
[ 00:00:03.011 ]: [Info] Loading scene...
[ 00:00:03.121 ]: [Info] Scene loaded in 110 ms
[ 00:00:03.128 ]: [Error] Invalid buffer size.
[ 00:00:03.129 ]: [Warning] Failed to get mesh data.
[ 00:00:03.130 ]: [Warning] E:\Flax/Projects/TestProject/Output/Windows/Content/SceneData/Scene/GrassCube.flax (Model)
[ 00:00:03.156 ]: [Warning] Exception has been thrown during . Failed to download mesh data.
Stack strace:
   at FlaxEngine.Mesh.DownloadIndexBuffer(Boolean forceGpu) in E:\Flax\Source\Source\Engine\Graphics\Mesh.cs:line 624
   at FlaxEngine.Utilities.MeshDataCache.DownloadMeshData() in E:\Flax\Source\Source\Engine\Utilities\MeshDataCache.cs:line 122
[ 00:00:03.156 ]: [Warning] Exception has been thrown during .
Failed to download mesh data.
```

Disabling the camera script doesn't help, but as soon it's removed everything works again.
O_o

**Comment by pacyfist** [comment]
Additional Details.
It seems like the important thing is how long does the `OnFixedUpdate` takes.

This works fine:
```csharp
using FlaxEngine;
using System.Threading;

public class FreeCamera : Script
{
    public override void OnFixedUpdate()
    {
        Thread.Sleep(1);
    }
}
```

Sleeping for 5 milliseconds

```csharp
public class FreeCamera : Script
{
    public override void OnFixedUpdate()
    {
        Thread.Sleep(5);
    }
}
```

 causes the exception:
 
 ```
 [ 00:00:04.128 ]: [Error] Invalid buffer size.
[ 00:00:04.130 ]: [Warning] Failed to get mesh data.
[ 00:00:04.132 ]: [Warning] E:\Flax/Projects/TestProject/Output/Windows/Content/SceneData/Scene/GrassCube.flax (Model)
[ 00:00:04.158 ]: [Warning] Exception has been thrown during . Failed to download mesh data.
Stack strace:
   at FlaxEngine.Mesh.DownloadIndexBuffer(Boolean forceGpu) in E:\Flax\Source\Source\Engine\Graphics\Mesh.cs:line 624
   at FlaxEngine.Utilities.MeshDataCache.DownloadMeshData() in E:\Flax\Source\Source\Engine\Utilities\MeshDataCache.cs:line 122
[ 00:00:04.158 ]: [Warning] Exception has been thrown during .
Failed to download mesh data.
```

And sleeping for 100 milliseconds

```csharp
public override void OnFixedUpdate()
{
   Thread.Sleep(100);
}
```

 simply doesn't return any data:

```
[ 00:00:03.594 ]: [Info] LOAD Base Vertices Count = 0
[ 00:00:03.594 ]: [Info] LOAD Base Triangles Count = 0
```

**Comment by mafiesto4** [comment]
The issue was caused by using Mesh data (Vertices/Indices count) before it's streamed which should not affect manual API to download that data. Fixed in 123b7f5433599b078710eb8f7800b61e9300b207
